model/project_ref.go

```diff
@@ -2203,6 +2242,40 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 	return res, nil
 }
 
+// ValidateProjectCreation returns an error if the total or owner/repo project limit set by the admin settings
+// has been reached. If the project is not enabled, it will grip a warning but not error.
+func ValidateProjectCreation(config *evergreen.Settings, projectRef *ProjectRef) error {
+	catcher := grip.NewBasicCatcher()
+	allEnabledProjects, err := GetNumberOfEnabledProjects()
+	if err != nil {
+		return errors.Wrap(err, "getting number of projects")
+	}
+	if allEnabledProjects >= config.ProjectCreation.TotalProjectLimit {
+		catcher.Errorf("total project limit of %d reached", config.ProjectCreation.TotalProjectLimit)
+	}
+	if !config.ProjectCreation.IsExceptionToRepoLimit(projectRef.Owner, projectRef.Repo) {
+		enabledOwnerRepoProjects, err := GetNumberOfEnabledProjectsForOwnerRepo(projectRef.Owner, projectRef.Repo)
+		if err != nil {
+			return errors.Wrapf(err, "getting number of projects for '%s/%s'", projectRef.Owner, projectRef.Repo)
+		}
+		if enabledOwnerRepoProjects >= config.ProjectCreation.RepoProjectLimit {
```

Comment:
This function doesn't check that these things could be empty; if totalProjectLimit and RepoProjectLimit are empty we should return from this function early so that we don't just completely block project creation.